### PR TITLE
Optimization Pass on reagent code(fuck reaction code so hard)

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
code: Makes reagent code work a little faster.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
This is some serious cargo cult, do you have any profiles to prove this does anything other than rearrange code

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -21,6 +21,7 @@
 		//Chemical Reagents - Initialises all /datum/reagent into a list indexed by reagent id
 		var/paths = subtypesof(/datum/reagent)
 		GLOB.chemical_reagents_list = list()
+		for(var/i in 1 to paths.len)
```
Comment (Jalleo) [inline code/modules/reagents/chemistry/holder.dm]: Undefined paths

File: code/modules/reagents/chemistry/holder.dm [hunk 2]
```diff
@@ -264,17 +267,17 @@
 					if(R.overdosed)
 						need_mob_update += R.overdose_process(C)
 					if(is_type_in_list(R,cached_addictions))
-						for(var/addiction in cached_addictions)
-							var/datum/reagent/A = addiction
+						for(var/i in 1 to cached_addictions.len)
```
Comment (Jalleo) [inline code/modules/reagents/chemistry/holder.dm]: Must not be I I is already being used in this chain for the cached reagents.

**Review by Jalleo — CHANGES_REQUESTED** [review]
Things goof must do. Alongside the oranges statement.

**Comment by duncathan** [comment]
fix your shit goofball

**Pull request closed without merging**